R/sample_predictions.R

```diff
@@ -300,3 +345,31 @@ sample_nowcasts <- function(point_nowcast_matrix,
   draws_df <- Reduce(rbind, draws_df_list)
   return(draws_df)
 }
+#' Sample from negative binomial model given a set of predictions
+#'
+#' @param pred Vector of predictions. Default is `NULL`
+#' @param uncertainty_params Vector of uncertainty parameters. Default is
+#'     `NULL`.
+#' @importFrom stats rnbinom
+#' @export
+#' @returns `sampled_pred` Object of the same dimensions as `pred` representing
+#'    a single draw from the negative binomial distribution
+#'    with the specified `uncertainty params`.
+#' @examples
+#' pred <- c(3.2, 4.6)
+#' sampled_preds <- sample_nb(pred,
+#'   uncertainty_params = c(50, 100)
+#' )
+#' sampled_preds
+sample_nb <- function(pred = NULL, uncertainty_params = NULL) {
```

Comment:
these should likely go in their own file especially if there are going to be more of them at some point